### PR TITLE
Fixes Component Error Handling

### PR DESCRIPTION
**Pull request by manifestinteractive**
There was a try / catch block that passed all JS errors to a component that did not exist.  I updated this and also called a console.error to get a stack trace on the actual file and line that was is causing the problem.  Without the console.error no developer output is generated in the Docs if there was a problem with a specific component.

Also fixed the flag component to add the additional position properties that were missing, which causes warnings on every page view.

### CODE REVIEW

**Comment by manifestinteractive** [comment]
To test this, you can just "break" one of the existing components.

For example, you can change code in the file [buttonTemplate.jsx](https://github.com/holidayextras/ui-toolkit/blob/master/src/components/button/code/templates/buttonTemplate.jsx)

`className={classNames(classes)}`

to 

`cassName={classNames(classes)}`

then re-render the docs, and you will see both an error where the component was rendered, as well as in the developer console with a stack trace to the actual error.

**Comment by lukehansell-hx** [comment]
looks good to me :small_blue_diamond: glad we can now see the actual errors that's happening :fallen_leaf:

File: src/components/flag/code/views/flagView.jsx [hunk 1]
```diff
@@ -5,7 +5,20 @@ module.exports = React.createClass({
...
+      'bottom',
+      'left',
+      'top left',
+      'bottom left',
+      'top right',
+      'bottom right',
+      'left top',
+      'left bottom',
+      'right top',
+      'right bottom'
```
Comment (jodiedoubleday) [inline src/components/flag/code/views/flagView.jsx]: Can we add `'right left'` and `'left right'`.

File: docs/src/CustomComponent.jsx [hunk 2]
```diff
@@ -195,8 +195,12 @@ var CustomComponent = React.createClass({
       }
     } catch (err) {
       this.setTimeout(function() {
+        if(typeof console !== 'undefined')
+        {
+          console.error(err);
+        }
         React.render(
-          <UIToolkit.BootstrapAlert bsStyle="danger">{err.toString()}</UIToolkit.BootstrapAlert>,
+          <div className="alert alert-danger">{err.toString()}</div>,
```
Comment (jodiedoubleday) [inline docs/src/CustomComponent.jsx]: good spot :+1:

**Comment by jodiedoubleday** [comment]
Happy to go live once right left etc has been added

**Comment by jodiedoubleday** [comment]
looks good :+1:  happy for this to go live

**Pull request merged**